packages/@react-stately/collections/src/useCollection.ts

```diff
@@ -26,5 +26,5 @@ export function useCollection<T extends object, C extends Collection<Node<T>> =
     return prev.current;
   // Don't invalidate when any prop changes, just the two we care about.
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [builder, props.children, props.items, context]);
+  }, [builder, props.children, props.items, context, factory]);
```

Comment:
this breaks the Table tests, but I need to add this to invalidate the collection when the factory changes (aka when a filter function is being applied to the node). Looking for an alternate route atm

Reply:
An alternative idea might be to add a `prop.invalidator` to the dep array which would give users the ability to provide a value that when changed causes the collection to be recalculated?